replica/table.cc

```diff
@@ -115,17 +115,20 @@ table::make_sstable_reader(schema_ptr s,
 }
 
 sstables::sstable_set compaction_group::make_compound_sstable_set() {
-    return sstables::make_compound_sstable_set(_t.schema(), { _main_sstables, _maintenance_sstables });
+    return sstables::make_compound_sstable_set(_t.schema(), { &_main_sstables, &_maintenance_sstables });
```

Comment:
Don't we lose snapshot semantics with this?

This can cause compound_sstable_set::for_each_sstable_gently_until() to skip an sstable when it moves from the maintenance set to the main set, and we already walked over the main set.

Reply:
for_each_sstable_gently_until takes a snapshot of all sstable_set_impl:s and walks over the snapshot, exactly because of this reason.

Reply:
Right, this one is covered.